org.eclipse.eclemma.ui.test/src/org/eclipse/eclemma/ui/DumpExecutionDataTest.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ ******************************************************************************/
+package org.eclipse.eclemma.ui;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.core.runtime.ILogListener;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationType;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.eclemma.core.CoverageTools;
+import org.eclipse.eclemma.core.ISessionManager;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable;
+import org.eclipse.swtbot.swt.finder.results.VoidResult;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class DumpExecutionDataTest {
+
+	private static final SWTWorkbenchBot bot = new SWTWorkbenchBot();
+
+	private static ILaunch launch1;
+	private static ILaunch launch2;
+
+	@AfterClass
+	public static void resetWorkbench() {
+		bot.resetWorkbench();
+	}
+
+	@BeforeClass
+	public static void setup() throws Exception {
+		openCoverageView();
+
+		JavaProjectKit project = new JavaProjectKit();
+		project.enableJava5();
+		final IPackageFragmentRoot root = project.createSourceFolder();
+		final IPackageFragment fragment = project.createPackage(root, "example");
+		project.createCompilationUnit(fragment, "Example.java",
+				"package example; class Example { public static void main(String[] args) throws Exception { Thread.sleep(30000); } }");
+
+		JavaProjectKit.waitForBuild();
+
+		ILaunchConfiguration launchConfiguration = createLaunchConfiguration(project.javaProject, "example.Example");
+
+		launch1 = launchConfiguration.launch(CoverageTools.LAUNCH_MODE, null);
+		launch2 = launchConfiguration.launch(CoverageTools.LAUNCH_MODE, null);
+	}
+
+	@Test
+	public void test() throws Exception {
+		final LogListener logListener = new LogListener();
+		Platform.addLogListener(logListener);
+
+		SWTBotView coverageView = bot.viewByTitle("Coverage");
+		coverageView.show();
+
+		coverageView.toolbarButton("Dump Execution Data").click();
+		bot.shell("Dump Execution Data").activate();
+		bot.button("Cancel").click();
+
+		coverageView.toolbarButton("Dump Execution Data").click();
+		bot.shell("Dump Execution Data").activate();
+		bot.button("OK").click();
```

Comment:
@marchof I've been thinking about this:

First of all - given that dump happens asynchronously, simplest option will be check after some timeout. But maybe you have an idea of a better option?

Secondly - as I noted in https://bugs.eclipse.org/bugs/show_bug.cgi?id=519149 :

> open "Dump Execution Data" dialog, notice that on Linux one element will be selected by default, while nothing is selected on Mac OS X and Windows

Question "why no default selection" IMO is a different story. So as of today there won't be dump on Mac OS X and Windows when "OK" immediately pressed without selection. And test asserts only that there is no exceptions and formally this is exactly what is required for resolution of bug reports. Yesterday it took me some time to prepare only this, decided to not dig deeper since night came.

Probably better to add assertions at least to keep information about this difference between OSes.